File: AdminUi/packages/admin_api_types/lib/src/identities/identity_deletion_process.dart
import 'package:admin_api_types/src/admin_api_types_base.dart';
import 'package:json_annotation/json_annotation.dart';

part 'identity_deletion_process.g.dart';

@JsonSerializable()
class IdentityDeletionProcess {
  final String id;
  final String status;
  final DateTime createdAt;
  final DateTime approvalPeriodEndsAt;
  final DateTime? approvalReminder1SentAt;
  final DateTime? approvalReminder2SentAt;
  final DateTime? approvalReminder3SentAt;
  final DateTime? approvedAt;
  final String? approvedByDevice;
  final DateTime? gracePeriodEndsAt;
  final DateTime? gracePeriodReminder1SentAt;
  final DateTime? gracePeriodReminder2SentAt;
  final DateTime? gracePeriodReminder3SentAt;

  IdentityDeletionProcess({
    required this.id,
    required this.status,
    required this.createdAt,
    required this.approvalPeriodEndsAt,
    this.approvalReminder1SentAt,
    this.approvalReminder2SentAt,
    this.approvalReminder3SentAt,
    this.approvedAt,
    this.approvedByDevice,
    this.gracePeriodEndsAt,
    this.gracePeriodReminder1SentAt,
    this.gracePeriodReminder2SentAt,
    this.gracePeriodReminder3SentAt,
  });

  factory IdentityDeletionProcess.fromJson(dynamic json) => _$IdentityDeletionProcessFromJson(json as Map<String, dynamic>);
  Map<String, dynamic> toJson() => _$IdentityDeletionProcessToJson(this);
}

@JsonSerializable()
class IdentityDeletionProcessDetail {
  final String id;
  final String status;
  final DateTime createdAt;
  final List<IdentityDeletionProcessAuditLogEntry> auditLog;

  IdentityDeletionProcessDetail({
    required this.id,
    required this.status,
    required this.createdAt,
    required this.auditLog,
  });

  factory IdentityDeletionProcessDetail.fromJson(dynamic json) => _$IdentityDeletionProcessDetailFromJson(json as Map<String, dynamic>);
  Map<String, dynamic> toJson() => _$IdentityDeletionProcessDetailToJson(this);
}

File: Sdks/ConsumerApi.Sdk/src/Endpoints/Identities/Types/IdentityDeletionProcess.cs
namespace Backbone.ConsumerApi.Sdk.Endpoints.Identities.Types;

public class IdentityDeletionProcess
{
    public required string Id { get; set; }
    public required List<IdentityDeletionProcessAuditLogEntry> AuditLog { get; set; }
    public required string Status { get; set; }
    public DateTime CreatedAt { get; set; }
    public required string CreatedByDevice { get; set; }

    public DateTime? GracePeriodEndsAt { get; set; }

    public DateTime? GracePeriodReminder1SentAt { get; set; }
    public DateTime? GracePeriodReminder2SentAt { get; set; }
    public DateTime? GracePeriodReminder3SentAt { get; set; }
}

public class IdentityDeletionProcessAuditLogEntry
{
    public required string Id { get; set; }
    public DateTime CreatedAt { get; set; }
    public required string Message { get; set; }
    public string? OldStatus { get; set; }
    public required string NewStatus { get; set; }
}
